symfony/src/Repository/EventRepository.php

```diff
@@ -51,4 +54,23 @@ public function getRecent(array $types = []): array
             ->enableResultCache(3600)
             ->getResult();
     }
+
+    public function getLatestEventTimestamp(): ?DateTimeImmutable
+    {
+        try {
+            $resultData = $this
+                ->createQueryBuilder('e')
+                ->select('MAX(e.timestamp)')
+                ->getQuery()
+                ->enableResultCache(3600)
+                ->getSingleScalarResult();
+            if (null === $resultData) {
+                return null;
+            }
+
+            return UtcClock::at(Enforce::nString($resultData));
+        } catch (NoResultException) {
+            return null;
+        }
```

Comment:
So... sorry for my mistake, the exception would not be thrown in this case. You probably can switch to `Enforce::string`, because null won't happen after line 67. I am adding an issue to improve phpdocs/interface of UtcClock, because I have no idea what was I thinking allowing null/false and returning current time in that case.